pumpkin/src/client/client_packet.rs

```diff
@@ -151,6 +151,14 @@ impl Client {
         };
 
         if BASIC_CONFIG.online_mode {
+            // Don't allow duplicate UUIDs
+            if let Some(online_player) = &server.get_player_by_uuid(profile.id).await {
+                log::debug!("Player (IP '{}', username '{}') tried to log in with the same UUID ('{}') as an online player (IP '{}', username '{}')", &self.address.lock().await.to_string(), &profile.name, &profile.id.to_string(), &online_player.client.address.lock().await.to_string(), &online_player.gameprofile.name);
+                self.kick("You are already connected to this server").await;
+                return;
+            }
```

Comment:
Check for duplicate uuid when authentication is successful.

Reply:
Im thinking of even check for duplicate uuid if online mode is true **or** false. Thats what Mojang is doing.

Reply:
marking as resolved

Reply:
Should be resolved in [cded477](https://github.com/Snowiiii/Pumpkin/pull/178/commits/cded477f0e149a12033c7786aebf67ac7fee6bd1) 🙂